routes/cmview.js

```diff
@@ -859,86 +860,53 @@ router.post("/:cmid/cls/:clid/convos", function (req, res) {
   var cmid = req.body.cmid;
   var clid = req.body.clid;
   var subject = req.body.subject;
+  var content = req.body.content;
+  var commid = req.body.commid;
 
   if (Number(cmid) !== Number(req.user.cmid)) {
     req.flash("warning", "Mixmatched user cmid and request user cmid insert.");
     res.redirect(redirect_loc);
   } else {
 
-    // close all the other conversations
-    db("convos")
-    .where("client", clid)
-    .andWhere("cm", cmid)
-    .andWhere("convos.open", true)
-    .pluck("convid")
-    .then(function (convos) {
-      
-      db("convos").whereIn("convid", convos)
-      .update({
-        open: false
-      }).then(function (success) {
-        
-        db("convos")
-        .insert({
-          cm: cmid,
-          client: clid,
-          subject: subject,
-          open: true,
-          accepted: true
-        }).returning("convid").then(function (convids) {
-
-          var convid = convids[0];
-          var content = req.body.content;
-          var commid = req.body.commid;
-
-          db("comms")
-          .where("commid", commid)
-          .limit(1)
-          .then(function (comms) {
-            
-            if (comms.length > 0) {
-              var comm = comms[0];
-
-              twClient.sendSms({
-                to: comm.value,
-                from: TWILIO_NUM,
-                body: content
-              }, function (err, msg) {
-                if (err) {
-                  console.log("Twilio send error: ", err);
-                  if (err.hasOwnProperty("code") && err.code == 21211) res.status(500).send("That number is not a valid phone number.");
-                  else res.redirect("/500");
-                } else {
-                  db("msgs")
-                  .insert({
-                    convo: convid,
-                    comm: commid,
-                    content: content,
-                    inbound: false,
-                    read: true,
-                    tw_sid: msg.sid,
-                    tw_status: msg.status
-                  })
-                  .returning("msgid")
-                  .then(function (msgs) {
-
-                    req.flash("success", "New conversation created.");
-                    redirect_loc = redirect_loc + "/convos/" + convid;
-                    res.redirect(redirect_loc);
-
-                  }).catch(errorRedirect);
-                }
-              });
-
-            } else { res.redirect("/500"); }
+    var newConvoId;
+
+    Convo.closeAll(cmid, clid)
+    .then(() => {
+      return Convo.create(cmid, clid, subject, true)
+    }).then((convoId) => {
+      newConvoId = convoId
+      return Communication.findById(commid)
+    }).then((communication) => {
+      twClient.sendSms({
```

Comment:
As we're now avoiding nested functions, this pattern might also be useful to look at:
https://github.com/caolan/async#waterfalltasks-callback

Gets passed an array of functions and supports error handling. Wildly popular apparently. Avoids nesting and futures. Worth a look.